activerecord/test/cases/base_test.rb

```diff
@@ -691,16 +699,111 @@ def test_comparison_with_different_objects_in_array
   end
 
   def test_readonly_attributes
-    assert_equal Set.new([ "title", "comments_count" ]), ReadonlyTitlePost.readonly_attributes
+    assert_equal Set.new([ "title" ]), ReadonlyTitlePost.readonly_attributes
 
     post = ReadonlyTitlePost.create(title: "cannot change this", body: "changeable")
+    assert_equal "cannot change this", post.title
+    assert_equal "changeable", post.body
+
+    post = Post.find(post.id)
+    assert_equal "cannot change this", post.title
+    assert_equal "changeable", post.body
+
+    assert_raises(ActiveRecord::ReadonlyAttributeError) do
+      post.title = "changed via write_attribute"
+    end
+    post.body = "changed via write_attribute"
+    assert_equal "changed via write_attribute", post.body
+
+    assert_raises(ActiveRecord::ReadonlyAttributeError) do
+      post.assign_attributes(body: "changed via assign_attributes", title: "changed via assign_attributes")
+    end
+    assert_equal "cannot change this", post.title
+    assert_equal "changed via assign_attributes", post.body
+
+    assert_raises(ActiveRecord::ReadonlyAttributeError) do
+      post.update(title: "changed via update", body: "changed via update")
```

Comment:
I don't see a test for this, so that is why I'm asking but are we allowing `write_attribute(:title, "some title")?

Reply:
Ah, yeah that was an oversight. I thought `post.title =` was using `write_attribute` based on the test but it uses `_write_attribute`.

I've added tests for `write_attribute` and fixed `HasReadonlyAttributes` to override that instead of `[]=`